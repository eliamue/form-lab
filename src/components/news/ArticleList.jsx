import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => (
  <ul aria-label="articles">
    {articles.map(({ title, author, description }) => (
      <li key={`${title}-${author}`}>
        <Article title={title} author={author} description={description} />
      </li>
    ))}
  </ul>
);

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      author: PropTypes.string,
      description: PropTypes.string,
    })
  ),
};

export default ArticleList;
