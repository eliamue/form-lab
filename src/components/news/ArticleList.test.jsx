import dotenv from 'dotenv';
dotenv.config();
import React from 'react';
import { render, screen } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList', () => {
  it('displays a list of articles', () => {
    render(
      <ArticleList
        articles={[
          {
            // eslint-disable-next-line max-len
            title: 'Stephen King quiere más películas de ‘Sharknado’ y ofrece un par de ideas',
            author: 'Hipersónica',
            // eslint-disable-next-line max-len
            description: 'Quizá un día toque hacer una reivindicación sentida de las películas de Sharknado. Quizás. De lo que no cabe duda es que las chifladas películas de desastres no le hacen ningún mal a nadie e incluso le pueden salvar una tarde de domingo muy aburrida. No sabem…',
          },
          {
            // eslint-disable-next-line max-len
            title: 'Sharknado: The Complete 6-Movie Saga in HD Digital for $4.99 at Microsoft',
            author: 'cheebong',
            // eslint-disable-next-line max-len
            description: 'Microsoft has Sharknado: The Complete Saga for $4.99. TODAY only. It is NOT Movies Anywhere, so no port. Own all 6 Sharknado films – the entire saga! Take the journey from the first storm, crushi ...',
          },
        ]}
      />
    );

    const ul = screen.getByRole('list', { name: 'articles' });
    expect(ul).toMatchSnapshot();
  });
});
