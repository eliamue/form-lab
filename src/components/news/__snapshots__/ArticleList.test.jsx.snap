// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList displays a list of articles 1`] = `
<ul
  aria-label="articles"
>
  <li>
    <div>
      <figure>
        <figcaption>
          Stephen King quiere más películas de ‘Sharknado’ y ofrece un par de ideas
           - 
          Hipersónica
        </figcaption>
        <p>
          Quizá un día toque hacer una reivindicación sentida de las películas de Sharknado. Quizás. De lo que no cabe duda es que las chifladas películas de desastres no le hacen ningún mal a nadie e incluso le pueden salvar una tarde de domingo muy aburrida. No sabem…
        </p>
      </figure>
    </div>
  </li>
  <li>
    <div>
      <figure>
        <figcaption>
          Sharknado: The Complete 6-Movie Saga in HD Digital for $4.99 at Microsoft
           - 
          cheebong
        </figcaption>
        <p>
          Microsoft has Sharknado: The Complete Saga for $4.99. TODAY only. It is NOT Movies Anywhere, so no port. Own all 6 Sharknado films – the entire saga! Take the journey from the first storm, crushi ...
        </p>
      </figure>
    </div>
  </li>
</ul>
`;
